### PR TITLE
Fix bookmark and replication key-properties

### PR DESCRIPTION
**Pull request by diegobatt**
# Description of change

Addressed the issues I created https://github.com/singer-io/tap-pendo/issues/34 and https://github.com/singer-io/tap-pendo/issues/33 

I basically remove the else in this control flow:
```
if record_timestamp > bookmark_dttm:
      singer.write_record(stream.tap_stream_id, transformed_record)
      counter.increment()
  else:
      singer.write_record(stream.tap_stream_id, transformed_record)
      counter.increment()
```

Since it does not make sense. `if A do B, else do B too`. This was causing some tables to be fully replicated every time (like accounts, visitors, etc).

I also added the day/hour as a key along with several others for the event-based tables, this new keys are necessary to keep track of all the events and not ignore some of them due to primary key constraints. 
This change follows the [pendo documentation for grouped events](https://developers.pendo.io/docs/?bash#events-grouped)
for `Events, FeatureEvents, PageEvents and TrackEvents` and [the documentation for ungrouped events](https://developers.pendo.io/docs/?bash#events-ungrouped) for `PollEvents and GuideEvents`

# Manual QA steps
 - Discover the catalog `tap-pendo --config tap_config.json --catalog catalog.json > state.json`
 - Run `pylint tap_pendo -d missing-docstring -d logging-format-interpolation -d too-many-locals -d too-many-arguments`
```
   ************* Module tap_pendo
tap_pendo/__init__.py:104:2: W0511: TODO: sync code for stream goes here... (fixme)
************* Module tap_pendo.discover
tap_pendo/discover.py:53:4: R1705: Unnecessary "elif" after "return" (no-else-return)
tap_pendo/discover.py:109:8: C0103: Variable name "s" doesn't conform to snake_case naming style (invalid-name)
tap_pendo/discover.py:111:8: C0103: Variable name "s" doesn't conform to snake_case naming style (invalid-name)
************* Module tap_pendo.streams
tap_pendo/streams.py:236:0: C0301: Line too long (104/100) (line-too-long)
tap_pendo/streams.py:397:0: C0301: Line too long (116/100) (line-too-long)
tap_pendo/streams.py:449:0: C0301: Line too long (153/100) (line-too-long)
tap_pendo/streams.py:450:0: C0301: Line too long (149/100) (line-too-long)
tap_pendo/streams.py:474:0: C0301: Line too long (116/100) (line-too-long)
tap_pendo/streams.py:1025:0: C0301: Line too long (115/100) (line-too-long)
tap_pendo/streams.py:1:0: C0302: Too many lines in module (1135/1000) (too-many-lines)
tap_pendo/streams.py:317:19: C0103: Variable name "v" doesn't conform to snake_case naming style (invalid-name)
tap_pendo/streams.py:329:48: C0103: Variable name "f" doesn't conform to snake_case naming style (invalid-name)
tap_pendo/streams.py:522:8: C0103: Variable name "ts" doesn't conform to snake_case naming style (invalid-name)
tap_pendo/streams.py:662:8: C0103: Variable name "ts" doesn't conform to snake_case naming style (invalid-name)
tap_pendo/streams.py:747:8: C0103: Variable name "ts" doesn't conform to snake_case naming style (invalid-name)
tap_pendo/streams.py:998:12: C0103: Variable name "ts" doesn't conform to snake_case naming style (invalid-name)
************* Module tap_pendo.utils
tap_pendo/utils.py:12:0: C0103: Argument name "dt" doesn't conform to snake_case naming style (invalid-name)
tap_pendo/utils.py:16:0: C0103: Argument name "dt" doesn't conform to snake_case naming style (invalid-name)
tap_pendo/utils.py:21:4: C0103: Argument name "fn" doesn't conform to snake_case naming style (invalid-name)
tap_pendo/utils.py:27:16: C0103: Variable name "t0" doesn't conform to snake_case naming style (invalid-name)
tap_pendo/utils.py:28:16: C0103: Variable name "t" doesn't conform to snake_case naming style (invalid-name)
tap_pendo/utils.py:41:0: C0103: Argument name "l" doesn't conform to snake_case naming style (invalid-name)
tap_pendo/utils.py:41:0: C0103: Argument name "n" doesn't conform to snake_case naming style (invalid-name)
tap_pendo/utils.py:51:23: C0103: Variable name "f" doesn't conform to snake_case naming style (invalid-name)
tap_pendo/utils.py:59:0: C0103: Argument name "dt" doesn't conform to snake_case naming style (invalid-name)

-----------------------------------
Your code has been rated at 9.64/10
```
 - Check the tap is working `tap-pendo --config tap_config.json --catalog catalog.json | singer-check-tap > state.json`
 
```
cat state.json
Checking stdin for valid Singer-formatted data
The output is valid.
It contained 32240 messages for 12 streams.

     16 schema messages
  31279 record messages
    945 state messages

Details by stream:
+----------------+---------+---------+
| stream         | records | schemas |
+----------------+---------+---------+
| accounts       | 966     | 1       |
| features       | 14      | 1       |
| feature_events | 2659    | 2       |
| guides         | 8       | 1       |
| guide_events   | 0       | 2       |
| pages          | 1       | 1       |
| page_events    | 2572    | 2       |
| visitors       | 104     | 1       |
| events         | 568     | 1       |
| poll_events    | 0       | 1       |
| track_types    | 0       | 1       |
| track_events   | 24387   | 2       |
+----------------+---------+---------+
```

# Risks
 - 
 
# Rollback steps
 - revert this branch

### CODE REVIEW

**Comment by cmerrick** [comment]
Hi @diegobatt, thanks for your contribution!

In order for us to evaluate and accept your PR, we ask that you [sign a contribution license agreement](https://stitch-cla-enforcer.herokuapp.com/). It's all electronic and will take just minutes.

**Comment by cmerrick** [comment]
You did it @diegobatt!

Thank you for signing the Singer Contribution License Agreement.

File: tap_pendo/streams.py [hunk 1]
```diff
@@ -674,7 +692,7 @@ def get_body(self, period, first):
 class PollEvents(Stream):
     replication_method = "INCREMENTAL"
     name = "poll_events"
-    key_properties = ['visitor_id', 'account_id', 'server_name', 'remote_ip']
+    key_properties = []
```
Comment (leslievandemark) [inline tap_pendo/streams.py:695]: Hello, thank you for the contribution. Could you provide us with more information about the reason for this change to `key_properties = []`, for both the `PollEvents` and `GuideEvents` streams, and how you tested them?

Comment (diegobatt) [reply tap_pendo/streams.py:695]: Hi! @leslievandemark  , I followed the documentation on ungrouped events https://developers.pendo.io/docs/?bash#events-ungrouped that both PollEvents and GuideEvents below to. For instance in GuideEvents documentation, the first bullet says:

```
Returns one event row for every guide event that occurs. This is not a summary or bucket structure like previous examples.
```

So there is no aggregation involved, so there shouldn't be unique constraints on any keys.

I tested them running the tap, although I didn't have guides or polls to retrieve in the first place

**Comment by diegobatt** [comment]
Is there a way you could review this PR @leslievandemark ? this has been sitting here for almost 3 months.

If you consider this is not production-ready, I don't mind putting some extra work into it. This issue is affecting our integration in stitch and it will be very helpful to have it fixed, I will be happy to address any comments or concerns on the changes that you might have

**Comment by Vi6hal** [comment]
looks like the team has addressed this issue in this PR #88 ,
closing this PR as cleaning up old PR's

**Review by luandy64 — APPROVED** [review]

**Pull request closed without merging**